packages/unmock-core/src/service/dsl/dsl.ts

```diff
@@ -33,16 +28,20 @@ export abstract class DSL {
    * @param schema
    * @returns A translated list of arguments.
    */
-  public static translateDSLToOAS(state: any, schema: Schema): any {
-    let translated: { [OASKey: string]: string | number | boolean } = {};
-    if (state.$size !== undefined) {
-      throwOnErrorIfStrict(() => {
-        translated = { ...translated, ...translate$size(state, schema) };
-        delete state.$size;
-      });
-    }
-    return translated;
+  public static replaceDSLWithOAS(state: any, schema: Schema): any {
```

Comment:
Ditto, easy candidate for removing the `delete`.

Reply:
This function is a bit hard for me to understand 🤔 It's modifying the input object by deleting properties from it and at the same time it's also returning stuff. I think it would be better to have either-or: either only modify the input and return `void`, or make it a pure function returning a new clone of the object, possibly along with the list of translated arguments if that's needed.

Reply:
Yeah that's fair. Will look into it (require changes elsewhere actually, but could be a code smell).

Reply:
Will be a different PR though.